### ISSUE TITLE
Override new Span/Buffer-based APIs on SslStream

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#22398]
Dependent on #22381.

- [ ] Override Read(Span)
- [ ] Override Write(ReadOnlySpan)
- [ ] Override ReadAsync(Memory)
- [x] Override WriteAsync(ReadOnlyMemory) (https://github.com/dotnet/corefx/pull/24295)

**Comment by Drawaes** [comment]
Where are we at with being able to go ahead with this? I see the stream changes are merged. I would like to use it as a basis for some ideas I have. I am happy to do the work if there is an example stream I can copy.

**Comment by stephentoub** [comment]
> Where are we at with being able to go ahead with this?

The span-based overloads can be added.  The buffer-based ones can't be done yet as buffer isn't in coreclr/corefx yet.

**Comment by Drawaes** [comment]
Is there an approx eta on the Buffer coming into Coreclr/corefx?

**Comment by stephentoub** [comment]
> Is there an approx eta on the Buffer coming into Coreclr/corefx?

@ahsonkhan, @KrzysztofCwalina, @joshfree, can you comment?

**Comment by joshfree** [comment]
We're going to start moving Buffer over this week.  @shiftylogic can you comment on ETA?

**Comment by Drawaes** [comment]
Any update to this as the week has gone?

**Comment by ahsonkhan** [comment]
@stephentoub, @Drawaes, assuming all goes smoothly, we are aiming to have Buffer\<T\> in coreclr/corefx by the end of this week. We have a few API adjustments left that should be done soon.

**Comment by stephentoub** [comment]
Thanks.

**Comment by Drawaes** [comment]
👍    Cool now just need @stephentoub to do the stream api change and I will look at SSLStreams implementation.

**Comment by stephentoub** [comment]
You mean on the base class? It's already done :)

**Comment by Drawaes** [comment]
Cool. Guess I need to put down my IL tool for a while and get back to SSLStream :)

**Comment by stephentoub** [comment]
@Drawaes, is this completed now?

**Comment by Drawaes** [comment]
Yes, the public API overrides are done for both read and write. Of course there is always more internal work to be done but that is not being tracked by this issue :)
